File: ClassSignIn/ClassSignIn/ClassViewController.m
//
//  ClassViewController.m
//  ClassSignIn
//
//  Created by tmachc on 15/11/26.
//  Copyright © 2015年 tmachc. All rights reserved.
//

#import "ClassViewController.h"

@interface ClassViewController ()

@end

@implementation ClassViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.automaticallyAdjustsScrollViewInsets = NO;
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

#pragma mark - table

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return 8;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 100;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *cellIdentifier = @"classCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
    if (!cell) {
        NSLog(@"nil--->>>");
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
        cell.selectionStyle = UITableViewCellSelectionStyleGray;
        cell.backgroundColor = [UIColor clearColor];
    }
    else {
//        for (UILabel *lab in cell.contentView.subviews) {
//            NSLog(@"lab--->>>%@",lab);
//            if (lab.tag == 1) {
//                lab.text = @"第一堂课";
//            }
//            else if (lab.tag == 2) {
//                lab.text = @"付老师";
//            }
//        }
    }
    
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [tableView deselectRowAtIndexPath:indexPath animated:NO];
}

@end

File: ClassSignInSwift/ClassSignInSwift/Class/ClassViewController.swift
//
//  ClassViewController.swift
//  ClassSignInSwift
//
//  Created by tmachc on 15/11/30.
//  Copyright © 2015年 tmachc. All rights reserved.
//

import UIKit

class ClassViewController: UIViewController, UITableViewDataSource, UITableViewDelegate{
    
    @IBOutlet var table: UITableView!
    
    var refreshControl: UIRefreshControl!
    
    var arrClassData = [Dictionary<String, String>]()

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        self.refreshControl = UIRefreshControl.init()
        self.refreshControl.addTarget(self, action: #selector(getClassData), forControlEvents: UIControlEvents.ValueChanged)
        self.table.addSubview(self.refreshControl)
    }
    
    // ********* MARK: - function
    
    func getClassData() {
        HttpManager.defaultManager.getRequest(
            url: HttpUrl,
            params: [
                "command": "getClassList",
                "userId": userDefault.objectForKey("_id")!,
                "type": userDefault.objectForKey("type")!
            ])
        { (result) -> Void in
            if result["code"]!.isEqual(0) {
                self.arrClassData = result["list"] as! [Dictionary<String, String>]
                self.table.reloadData()
            }
            else {
                ShowAlert(target: self, message: result["message"] as! String)
            }
            self.refreshControl.endRefreshing()
        }
    }
    
    // ********* MARK: - table
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return arrClassData.count
    }
    
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        return 100
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell: ClassTableViewCell = tableView.dequeueReusableCellWithIdentifier("classCellID") as! ClassTableViewCell
        cell.dicClassData = arrClassData[indexPath.row]
        return cell
    }

    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        self.performSegueWithIdentifier("oneClass", sender: indexPath)
    }
    
    // ********* MARK: - Navigation
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if segue.identifier == "oneClass" {
            let destinationController = segue.destinationViewController as! OneClassController
            destinationController.dicClassData = arrClassData[sender!.row]
        }
    }
    
    func addClass(sender: UIButton) {
        self.performSegueWithIdentifier("addNewClass", sender: nil)
    }
}
